lib/block-supports/index.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Block support flags.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Filter the registered blocks to apply the block supports attributes registration.
+ */
+function gutenberg_register_block_supports() {
+	$block_registry         = WP_Block_Type_Registry::get_instance();
+	$registered_block_types = $block_registry->get_all_registered();
+	// Ideally we need a hook to extend the block registration
+	// instead of mutating the block type.
+	foreach ( $registered_block_types as $block_type ) {
+		gutenberg_register_alignment_support( $block_type );
+	}
+}
+
+add_action( 'init', 'gutenberg_register_block_supports', 11 );
+
+/**
+ * Filters the frontend output of blocks and apply the block support flags transformations.
+ *
+ * @param  string $block_content rendered block content.
+ * @param  array  $block block object.
+ * @return string filtered block content.
+ */
+function gutenberg_apply_block_supports( $block_content, $block ) {
+	if ( ! isset( $block['attrs'] ) ) {
+		return $block_content;
+	}
+
+	$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	// If no render_callback, assume styles have been previously handled.
+	if ( ! $block_type || ! $block_type->render_callback ) {
+		return $block_content;
+	}
+
+	$attributes = array();
+	$attributes = gutenberg_apply_colors_support( $attributes, $block['attrs'], $block_type );
```

Comment:
We directly pass the block type to these functions, we don't need to manipulate the supports key here.